snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/TrackerWebViewInterfaceV2.kt

```diff
@@ -0,0 +1,140 @@
+
+import android.webkit.JavascriptInterface
+import com.snowplowanalytics.core.event.WebViewReader
+import com.snowplowanalytics.core.tracker.Logger.e
+import com.snowplowanalytics.core.utils.JsonUtils.jsonToMap
+import com.snowplowanalytics.snowplow.Snowplow.defaultTracker
+import com.snowplowanalytics.snowplow.Snowplow.getTracker
+import com.snowplowanalytics.snowplow.event.*
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.*
+
+/**
+ * JavaScript interface used to provide an API for tracking events from WebViews.
+ */
+class TrackerWebViewInterfaceV2 {
+    @JavascriptInterface
+    @Throws(JSONException::class)
+    fun trackWebViewEvent(
+        eventName: String,
+        trackerVersion: String,
+        useragent: String,
+        selfDescribingEventData: String? = null,
+        pageUrl: String? = null,
+        pageTitle: String? = null,
+        referrer: String? = null,
+        category: String? = null,
+        action: String? = null,
+        label: String? = null,
+        property: String? = null,
+        value: Double? = null,
+        pingXOffsetMin: Int? = null,
+        pingXOffsetMax: Int? = null,
+        pingYOffsetMin: Int? = null,
+        pingYOffsetMax: Int? = null,
+        entities: String? = null,
+        trackers: Array<String>? = null
+    ) {
+        val event = WebViewReader(
+            eventName,
+            trackerVersion,
+            useragent,
+            parseSelfDescribingEventData(selfDescribingEventData),
+            pageUrl,
+            pageTitle,
+            referrer,
+            category,
+            action,
+            label,
+            property,
+            value,
+            pingXOffsetMin,
+            pingXOffsetMax,
+            pingYOffsetMin,
+            pingYOffsetMax
+        )
+        trackEvent(event, entities, trackers)
+    }
+    
+    @Throws(JSONException::class)
+    private fun trackEvent(event: AbstractEvent, contextEntities: String?, trackers: Array<String>?) {
+        if (contextEntities != null) {
+            val entities = parseEntities(contextEntities)
+            if (entities.isNotEmpty()) {
+                event.entities(entities)
+            }
+        }
+        if (trackers.isNullOrEmpty()) {
+            val tracker = defaultTracker
+            if (tracker != null) {
+                tracker.track(event)
+            } else {
+                e(TAG, "Tracker not initialized.")
+            }
+        } else {
+            for (namespace in trackers) {
+                val tracker = getTracker(namespace)
+                if (tracker != null) {
+                    tracker.track(event)
+                } else {
+                    e(TAG, String.format("Tracker with namespace %s not found.", namespace))
+                }
+            }
+        }
+    }
+
+    @Throws(JSONException::class)
```

Comment:
Nit: is this throwing a JSON exception? I don't see any JSON parsing in the function.